src/instructlab/taxonomy/diff.py

```diff
@@ -0,0 +1,83 @@
+import click
+from instructlab import helpers
+import logging
+from instructlab.data import generator
+import yaml
+from git import GitError
+from instructlab import utils
+import os
+
+@click.command()
+@click.option(
+    "--taxonomy-path",
+    type=click.Path(),
+    help=f"Path to {helpers.DEFAULT_TAXONOMY_REPO} clone or local file path.",
+)
+@click.option(
+    "--taxonomy-base",
+    help="Base git-ref to use for taxonomy.",
+)
+@click.option(
+    "--yaml-rules",
+    type=click.Path(),
+    default=None,
+    help="Custom rules file for YAML linting.",
+)
+@click.option(
+    "--quiet",
+    is_flag=True,
+    help="Suppress all output. Call returns 0 if check passes, 1 otherwise.",
+)
+@click.pass_context
+def diff(ctx, taxonomy_path, taxonomy_base, yaml_rules, quiet):
+    """
+    Lists taxonomy files that have changed since <taxonomy-base>
+    and checks that taxonomy is valid. Similar to 'git diff <ref>'.
+    """
+    # pylint: disable=C0415
+    # Local
+    from ..utils import get_taxonomy_diff, read_taxonomy
+
+    if not taxonomy_base:
+        taxonomy_base = ctx.obj.config.generate.taxonomy_base
+    if not taxonomy_path:
+        taxonomy_path = ctx.obj.config.generate.taxonomy_path
+    if not ctx.obj:
+        logger = logging.getLogger(__name__)
+    else:
+        logger = ctx.obj.logger
+
+    if not quiet:
+        is_file = os.path.isfile(taxonomy_path)
+        if is_file:  # taxonomy_path is file
+            click.echo(taxonomy_path)
+        else:  # taxonomy_path is dir
+            try:
+                updated_taxonomy_files = get_taxonomy_diff(taxonomy_path, taxonomy_base)
+            except (SystemExit, GitError) as exc:
+                click.secho(
+                    f"Reading taxonomy failed with the following error: {exc}",
+                    fg="red",
+                )
+                raise SystemExit(1) from exc
+            for f in updated_taxonomy_files:
+                click.echo(f)
+    try:
+        read_taxonomy(logger, taxonomy_path, taxonomy_base, yaml_rules)
+    except (SystemExit, yaml.YAMLError) as exc:
+        if not quiet:
+            click.secho(
+                f"Reading taxonomy failed with the following error: {exc}",
+                fg="red",
+            )
+        raise SystemExit(1) from exc
+    if not quiet:
+        click.secho(
+            f"Taxonomy in /{taxonomy_path}/ is valid :)",
```

Comment:
```suggestion
            f"Taxonomy in /{taxonomy_path}/ is valid :)",
```
I've never understood why the slashes around `{taxonomy_path}` were added

Reply:
removed, I agree